### PR TITLE
[kube-prometheus-stack] Allow run Prometheus Operator Webhook receiver separately

### PR DESCRIPTION
**Pull request by jkroepke**
#### What this PR does / why we need it

Prometheus operator offers a dedicated image for the webhook receiver.

Running the Webhook receiver allows to run to run it in HA mode which is not supported by the Operator directly.

`--set prometheusOperator.admissionWebhooks.deployment.enabled=true` to enable it

Ref: https://github.com/prometheus-operator/prometheus-operator/issues/6110

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job-patch/job-createSecret.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ spec:
           imagePullPolicy: {{ .Values.prometheusOperator.admissionWebhooks.patch.image.pullPolicy }}
           args:
             - create
-            - --host={{ template "kube-prometheus-stack.operator.fullname" . }},{{ template "kube-prometheus-stack.operator.fullname" . }}.{{ template "kube-prometheus-stack.namespace" . }}.svc
+            - --host={{ template "kube-prometheus-stack.operator.fullname" . }},{{ template "kube-prometheus-stack.operator.fullname" . }}.{{ template "kube-prometheus-stack.namespace" . }}.svc{{- if .Values.prometheusOperator.admissionWebhooks.deployment.enabled }},{{ template "kube-prometheus-stack.operator.fullname" . }}-webhook,{{ template "kube-prometheus-stack.operator.fullname" . }}-webhook.{{ template "kube-prometheus-stack.namespace" . }}.svc{{- end }}
```
Comment (GMartinez-Sisti) [inline charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job-patch/job-createSecret.yaml:46]: What do you think of creating a template for this? It's kind of hard to read. I see that you have similar logic in other places, would be nice to centralize the logic.

Comment (jkroepke) [reply charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job-patch/job-createSecret.yaml]: What did you think about 9e8576b

**Comment by GMartinez-Sisti** [comment]
@jkroepke I think you reverted the wrong commit. It was suppose to remove the change on `certmanager.yaml` file, and not the new named template, right?

**Comment by jkroepke** [comment]
I see, I missunderstand you.

**Comment by jkroepke** [comment]
@GMartinez-Sisti it should cover everything now

**Review by GMartinez-Sisti — APPROVED** [review]

**Pull request merged**